sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -199,8 +199,9 @@ public void getAbsolutePath_with_no_deterministic_path_in_windows_path_returns_e
     verify(fs, never()).predicates();
     assertThat(logTester.logs(Level.TRACE)).isEmpty();
     assertThat(logTester.logs(Level.DEBUG)).hasSize(1);
-    assertThat(logTester.logs(Level.DEBUG).get(0)).isEqualTo("Did not find deterministic source path in 'C:\\_\\some\\path\\file.cs'." +
-      " Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.");
+    assertThat(logTester.logs(Level.DEBUG).get(0)).isEqualTo(
+      "The file 'C:\\_\\some\\path\\file.cs' is not indexed or does not have the supported language."
```

Comment:
this can fail either because there is a mismatch between the deterministic source path (due to some tooling issue) or because the file hasn't been indexed.

how can we make both possible cases clear? is it worth it? what would it help if you were the developer investigating such a case?

Did you look at the integration tests we have for deterministic code coverage, to get more context?

Reply:
I looked at the logging in `ScannerFileService` again, and the messages were plain wrong in some cases. I refactored it a bit to improve the situation. I hope the messages make sense now.